protocol/chainlib/chain_fetcher.go

```diff
@@ -44,6 +45,13 @@ func (cf *ChainFetcher) Validate(ctx context.Context) error {
 	return nil
 }
 
+func (cf *ChainFetcher) ChainFetcherMetadata() []pairingtypes.Metadata {
+	ret := []pairingtypes.Metadata{
```

Comment:
won't it be better if we store it in the struct and read if instead of every time allocating and accessing it?